ST_environs.c

```diff
@@ -60,27 +60,67 @@ void Env_Generate( void) {
   _make_disturbance();
 }
 
+/* Deep copy one EnvType's information to another. Note that both must be initialized
+   BEFORE calling this function. */
+void copy_environment(const EnvType* src, EnvType* dest){
+  if(!src){
+    return;
+  }
+
+  dest->lyppt = src->lyppt;
+  dest->ppt = src->ppt;
+  dest->temp = src->temp;
+  dest->gsppt = src->gsppt;
+  dest->temp_reduction[0] = src->temp_reduction[0];
+  dest->temp_reduction[1] = src->temp_reduction[1];
+  dest->wet_dry = src->wet_dry;
+}
+
+/* Deep copy one PlotType's information to another. Note that both must be initialized
+   BEFORE calling this function. */
+void copy_plot(const PlotType* src, PlotType* dest){
+  if(!src){
+    return;
+  }
+
+  dest->disturbance = src->disturbance;
+  dest->disturbed = src->disturbed;
+  dest->pat_removed = src->pat_removed;
+}
+
+/* Deep copy one SucculentType's information to another. Note that both must be initialized
+   BEFORE calling this function. */
+void copy_succulent(const SucculentType* src, SucculentType* dest){
+  if(!src){
+    return;
+  }
+
+  dest->growth[0] = src->growth[0];
+  dest->growth[1] = src->growth[1];
+  dest->mort[0] = src->mort[0];
+  dest->mort[1] = src->mort[1];
+  dest->prob_death = src->prob_death;
+  dest->reduction = src->reduction;
+}
+
 /**************************************************************/
 static void _make_ppt( void) {
 /*======================================================*/
 /* If not running SOILWAT,take a random number from normal distribution with*/
 /* mean, stddev that is between min & max from */
-/* the Globals.ppt structure.*/
-/* Also set the growing season precip. */
+/* the Globals->ppt structure.*/
+/* Also determine growing season precipitation. */
 
 /* HISTORY */
 /* Chris Bennett @ LTER-CSU 6/15/2000            */
 /* cwb - 6-Dec-02 -- added code to interface with STEPWAT.
  *       The ppt and gsppt are set in _sxw_set_environs()
  *       but we still pass through this code to set the
  *       Dry/Wet/Normal state.
- * KAP 1/26/2017 The above note by CB is not correct. Env.ppt
- *      is set in _sxw_set_environs, but gsppt is not, it is
- *      set below. When using SOILWAT or not, the gsspt, ppt.dry,
- *      and ppt.wet is currently fixed each year and read from env.in
- *      We should consider calculating gsspt in the _sxw_set_environs
- *      function when running SOILWAT (so it is not fixed), and allowing
- *      what constitutes a wet and dry year to vary across sites. */
+ * KAP 1/26/2017 We should consider calculating amount of growing-season 
+ *      precipitation this year in the _sxw_set_environs function when 
+ *      running SOILWAT (so it is not fixed), and allowing what constitutes 
+ *      a wet and dry year to vary across sites. */
```

Comment:
this comment needs to be deleted as gsppt is set down on the fly in _make_ppt

Reply:
Addressed by commit 6a53657.